libifupdown/interface.c

```diff
@@ -239,6 +240,12 @@ lif_interface_collection_inherit(struct lif_interface *interface, struct lif_dic
 	if (parent == NULL)
 		return false;
 
+	if (!lif_config.allow_any_iface_as_template && !parent->is_template)
+		return false;
+
+	/* explicitly convert any interface we are inheriting from into a template */
+	parent->is_template = true;
```

Comment:
This breaks setups where `allow_any_iface_as_template` is `true`, `parent` a real `iface` and should be used as such. The probability of this is low, but not 0.

I think this line should be removed.

Reply:
I turned this into a config option.